lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestCloseTask.cs

```diff
@@ -0,0 +1,171 @@
+            var messages = new List<string>();
+            messages.Add(DisposeMessageFromDriver);
+            ValidateMessageSuccessfullyLogged(messages, "Node-*", EvaluatorStdout, testFolder, 1);
+            CleanUp(testFolder);
+        }
+
+        /// <summary>
+        /// Driver configuration for the test driver
+        /// </summary>
+        /// <returns></returns>
+        public IConfiguration DriverConfigurations()
+        {
+            var helloDriverConfiguration = DriverConfiguration.ConfigurationModule
+                .Set(DriverConfiguration.OnDriverStarted, GenericType<StopTaskHandlers>.Class)
+                .Set(DriverConfiguration.OnEvaluatorAllocated, GenericType<StopTaskHandlers>.Class)
+                .Set(DriverConfiguration.OnContextActive, GenericType<StopTaskHandlers>.Class)
+                .Set(DriverConfiguration.OnTaskRunning, GenericType<StopTaskHandlers>.Class)
+                .Set(DriverConfiguration.OnTaskCompleted, GenericType<StopTaskHandlers>.Class)
+                .Build();
+
+            return TangFactory.GetTang().NewConfigurationBuilder(helloDriverConfiguration).Build();
+        }
+
+        private sealed class StopTaskHandlers :
+            IObserver<IDriverStarted>,
+            IObserver<IAllocatedEvaluator>,
+            IObserver<IActiveContext>,
+            IObserver<ICompletedTask>,
+            IObserver<IRunningTask>
+        {
+            private readonly IEvaluatorRequestor _requestor;
+            private int _contextNumber = 0;
+            private int _taskNumber = 0;
+
+            [Inject]
+            private StopTaskHandlers(IEvaluatorRequestor evaluatorRequestor)
+            {
+                _requestor = evaluatorRequestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().SetNumber(1).Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                value.SubmitContext(
+                    ContextConfiguration.ConfigurationModule
+                        .Set(ContextConfiguration.Identifier, "ContextID" + _contextNumber++)
+                        .Build());
+            }
+
+            public void OnNext(IActiveContext value)
+            {
+                value.SubmitTask(GetTaskConfiguration());
+            }
+
+            public void OnNext(ICompletedTask value)
+            {
+                // Log on task completion to signal a passed test.
+                Logger.Log(Level.Info, CompletedValidationMessage + "Task completed: " + value.Id);
+                value.ActiveContext.Dispose();
+            }
+
+            public void OnNext(IRunningTask value)
+            {
+                Logger.Log(Level.Info, "Task running: " + value.Id);
+                value.Dispose(Encoding.UTF8.GetBytes(DisposeMessageFromDriver));
```

Comment:
This only tests calls to `Dispose` with a message. It would be good to also test the case with no parameter.